packages/back-end/src/api/features/postFeature.ts

```diff
@@ -0,0 +1,155 @@
+import { z } from "zod";
+import { validateFeatureValue } from "shared/util";
+import { accountFeatures, getAccountPlan } from "enterprise";
+import { PostFeatureResponse } from "../../../types/openapi";
+import { createApiRequestHandler } from "../../util/handler";
+import { postFeatureValidator } from "../../validators/openapi";
+import { createFeature, getFeature } from "../../models/FeatureModel";
+import { getExperimentMapForFeature } from "../../models/ExperimentModel";
+import { FeatureInterface } from "../../../types/feature";
+import { getEnabledEnvironments } from "../../util/features";
+import {
+  addIdsToRules,
+  createInterfaceEnvSettingsFromApiEnvSettings,
+  getApiFeatureObj,
+  getSavedGroupMap,
+} from "../../services/features";
+import { auditDetailsCreate } from "../../services/audit";
+import { OrganizationInterface } from "../../../types/organization";
+
+export type ApiFeatureEnvSettings = NonNullable<
+  z.infer<typeof postFeatureValidator.bodySchema>["environments"]
+>;
+
+export type ApiFeatureEnvSettingsRules = ApiFeatureEnvSettings[keyof ApiFeatureEnvSettings]["rules"];
+
+export const validateEnvKeys = (
+  orgEnvKeys: string[],
+  incomingEnvKeys: string[]
+) => {
+  const invalidEnvKeys = incomingEnvKeys.filter((k) => !orgEnvKeys.includes(k));
+
+  if (invalidEnvKeys.length) {
+    throw new Error(
+      `Environment key(s) '${invalidEnvKeys.join(
+        "', '"
+      )}' not recognized. Please create the environment or remove it from your environment settings and try again.`
+    );
+  }
+};
+
+export const parseJsonSchemaForEnterprise = (
+  org: OrganizationInterface,
+  jsonSchema: string | undefined
+) => {
+  const defaultJsonSchema = {
+    schema: "",
+    date: new Date(),
+    enabled: false,
+  };
+  const commercialFeatures = [...accountFeatures[getAccountPlan(org)]];
+  if (!commercialFeatures.includes("json-validation")) return defaultJsonSchema;
+  try {
+    return jsonSchema ? JSON.parse(jsonSchema) : defaultJsonSchema;
```

Comment:
Instead of the user passing in an object like this (requiring nested JSON stringify and date/enabled):
```js
JSON.stringify({
  "schema": JSON.stringify(schema),
  date: new Date(),
  enabled: true
})
```

I think we can simplify it if they only pass in `JSON.stringify(schema)` itself.  We can auto-populate `date` and `enabled` for them.  I can't think of a good reason why someone would want to pass in a JSON schema, but set `enabled` to false

Reply:
Good catch, updated